### PR TITLE
Chore: Adds CI

### PR DESCRIPTION
**Pull request by vibalre**
* Adds `R CMD CHECK`
* Includes linting using `lintr` with settings specified in `.lintr`
* Adds unit test for `drawCell()` and tests for `drawCellShiny()` via `shinytest2`
* Checks for English spelling using `spelling` except for words listed in `inst/WORDLIST`
* Replaces single quotes with double quotes.
* Removes non-essential whitespaces.

### CODE REVIEW

**Comment by svalvaro** [comment]
This PR is huge enough to crash my Github lol ⚠️, I'm unable to review it since the page is loading quite slow.
I'll create a separate branch to add the latest changes that you've added.

File: tests/testthat/test-drawCell.R [hunk 1]
```diff
@@ -1,9 +1,15 @@
-test_that("Generates a picture", {
+test_that("drawCell enerates a picture", {
```
Comment (svalvaro) [inline tests/testthat/test-drawCell.R:1]: typo

File: tests/testthat/test-drawCell.R [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+  expect_s3_class(cell_svg, "htmlwidget")
+  expect_equal(cell_svg[["x"]][["input_id"]], "cell")
+  expect_equal(cell_svg[["x"]][["organism_identifier"]], "9060")
+  expect_equal(cell_svg[["x"]][["sl_ids"]], "SL0173,SL0101")
+  expect_equal(cell_svg[["x"]][["colour_sl"]], "red,blue")
 
-    drawCell::drawCell(taxonomy_id = '9606', sl_ids = '0101')
-
-
-    dir_cell <- paste0(getwd(),'/','cell_pic.png')
```
Comment (svalvaro) [inline tests/testthat/test-drawCell.R:6]: I'm glad that we are removing this old test :)

File: vignettes/drawCell.Rmd [hunk 3]
```diff
@@ -28,85 +26,53 @@ with highlighted subcellular organelles with R.
...
-Start by installing `webshot2` and Google Chrome (If you don't have it already).
-
-
-```{r installation1, eval = FALSE}
-remotes::install_github("rstudio/webshot2")
-```
-
-
-After installing it you can instdall `drawCell`:
+Install dev version of `drawCell`:
```
Comment (svalvaro) [inline vignettes/drawCell.Rmd:31]: Thanks for updating the vignettes!

**Review by svalvaro — APPROVED** [review]
Good job done here @vibalre, You have implemented many features. I'm approving since everything is working on my end.
Just let me give you some advice for future PR: Try to separate the work in smaller chunks. It will be easier for you and for the reviewer, (it took me 2 hours to go through it). Each of this features that you implement here, it could easily be its own branch and PR:

- Adds CI
- Update Vignettes & README & Description
- Updates testing
- Lint the package

But still, awesome job done here 💯

**Comment by vibalre** [comment]
@svalvaro Yes, sorry about the big pull request. I got too excited on this branch and put everything in one place. I'll make more bite sizes PRs next time.

**Pull request merged**